pkg/operation/botanist/controlplane.go

```diff
@@ -290,6 +291,19 @@ func (b *Botanist) HibernateControlPlane(ctx context.Context) error {
 		if err := c.Delete(ctx, &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.DeploymentNameKubeAPIServer, Namespace: b.Shoot.SeedNamespace}}, kubernetes.DefaultDeleteOptions...); client.IgnoreNotFound(err) != nil {
 			return err
 		}
+
+		fns := []flow.TaskFn{
+			func(ctx context.Context) error { return b.DestroyInternalDomainDNSRecord(ctx) },
+			func(ctx context.Context) error { return b.DestroyExternalDomainDNSRecord(ctx) },
+		}
+
+		if b.Shoot.NginxIngressEnabled() {
```

Comment:
Is this check needed? I think we would have a problem if a user changes in one shot the shoot spec to `hibernated.enabled=true`, `nginx-ingress.enabled=false`, or? Then the nginx-ingress DNS record would remain?

Reply:
Hm I think the extended condition `.DoIf(dnsEnabled && managedExternalDNS && !o.Shoot.HibernationEnabled)` prevents this.
https://github.com/gardener/gardener/blob/cf98a923f51ef3c2ea556032b46642d2869c0c09/pkg/gardenlet/controller/shoot/shoot_control_reconcile.go#L280

It's just a proposal to save some calls but since not found errors are ignored we can also execute it every time. I don't have a strong opinion.

Reply:
Yes nice catch. I'll remove it.